packages/airnode-adapter/src/response-processing/extraction.test.ts

```diff
@@ -66,11 +64,95 @@ describe('extractAndEncodeValue', () => {
+    const parameters: ReservedParameters = { _path: '', _type: 'int256' };
+    const res = extractAndEncodeResponse(data, parameters);
+    expect(res).toEqual({
+      rawValue: 123456,
+      values: ['123456'],
+      encodedValue: '0x000000000000000000000000000000000000000000000000000000000001e240',
+    });
+  });
+});
+
+describe('extract and encode multiple values', () => {
+  it('works for basic request', () => {
+    const data = { a: { b: [{ c: 1 }, { d: '750.51' }] } };
+    const parameters: ReservedParameters = { _path: 'a.b.1.d,a.b.0.c', _type: 'int256,bool', _times: '100,' };
+    const res = extractAndEncodeResponse(data, parameters);
+    expect(res).toEqual({
+      rawValue: data,
+      values: ['75051', true],
+      encodedValue:
+        '0x000000000000000000000000000000000000000000000000000000000001252b0000000000000000000000000000000000000000000000000000000000000001',
+    });
+  });
+
+  it('works for more complex request', () => {
+    const data = [12.3, 45.6];
+    const parameters: ReservedParameters = { _path: ',0', _type: 'int256[],int256', _times: '100,1000' };
+    const res = extractAndEncodeResponse(data, parameters);
+    expect(res).toEqual({
+      rawValue: data,
+      values: [['1230', '4560'], '12300'],
+      encodedValue:
+        '0x0000000000000000000000000000000000000000000000000000000000000040000000000000000000000000000000000000000000000000000000000000300c000000000000000000000000000000000000000000000000000000000000000200000000000000000000000000000000000000000000000000000000000004ce00000000000000000000000000000000000000000000000000000000000011d0',
+    });
+  });
+
+  it('correctly splits reserved parameters containing multiple values', () => {
+    expect(splitReservedParameters({ _type: 'uint256' })).toEqual([{ _type: 'uint256' }]);
+    expect(splitReservedParameters({ _type: 'uint256,string', _path: 'key,anotherKey' })).toEqual([
+      { _type: 'uint256', _path: 'key' },
+      { _type: 'string', _path: 'anotherKey' },
+    ]);
+    expect(splitReservedParameters({ _type: 'uint256, invalid' })).toEqual([
+      { _type: 'uint256' },
+      { _type: ' invalid' },
+    ]);
+    expect(splitReservedParameters({ _type: 'uint256,,,invalid' })).toEqual([
+      { _type: 'uint256' },
+      { _type: '' },
+      { _type: '' },
+      { _type: 'invalid' },
+    ]);
+    expect(splitReservedParameters({ _type: 'uint256,string,bytes', _path: 'usd,,' })).toEqual([
+      { _type: 'uint256', _path: 'usd' },
+      { _type: 'string', _path: '' },
+      { _type: 'bytes', _path: '' },
+    ]);
+    expect(splitReservedParameters({ _type: 'uint256', _times: '100' })).toEqual([{ _type: 'uint256', _times: '100' }]);
+  });
+
+  it('throws when there are different number of splits', () => {
+    expect(() => splitReservedParameters({ _type: 'uint256', _path: 'usd,eur' })).toThrow(
+      `Unexpected number of parsed reserved parameters. Number of "_types" parameters = 1, but "_path" has only 2`
```

Comment:
This should help catch some (hard to debug) issues if the API provider forgets to escape things.